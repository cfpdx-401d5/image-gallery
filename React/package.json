{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
