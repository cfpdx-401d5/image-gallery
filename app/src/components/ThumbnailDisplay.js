import React, { PropTypes } from 'react';

function ThumbnailDisplay(props) {
  return (
    <div className='image-divs'>
      <p>Title: {props.image.title} </p>
      <a href={props.image.url} > 
        <img className='thumbnail' alt='cute bunny' src={props.image.url} /> 
      </a>
      <button className='kill-button' onClick={() => props.deleteImage(props.image) }>
        Delete
      </button>      
    </div>
  );
}

ThumbnailDisplay.propTypes = {
  image: PropTypes.object.isRequired,
  deleteImage: PropTypes.func.isRequired,
};

export default ThumbnailDisplay;
