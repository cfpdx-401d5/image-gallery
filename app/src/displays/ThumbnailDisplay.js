import React, { PropTypes } from 'react';

import ThumbnailDisplayBits from './ThumbnailDisplayBits';

function ThumbnailDisplay(props) {
  const imagesAsTableRows = props.images.map(imageObj => {
    return <ThumbnailDisplayBits key={imageObj.id} title={imageObj.title} url={imageObj.url} deleteImage={props.deleteImage} id={imageObj.id} />;
  });

  return (
    <table>
        <tbody>
            {imagesAsTableRows}
        </tbody>
    </table>
  );
}

ThumbnailDisplay.propTypes = {
  deleteImage: PropTypes.func.isRequired, 
  images: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  })).isRequired,
};


export default ThumbnailDisplay;
