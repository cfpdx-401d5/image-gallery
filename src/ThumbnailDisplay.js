import React, { PropTypes } from 'react';

function ThumbnailDisplay(props) {
  return (
    <div className='image-divs'>
      <p> Title: {props.image.title} </p>
      <img className='thumbnail' src={props.image.url} alt='cute bunny'/>
    </div>
  );
}

ThumbnailDisplay.propTypes = {
  image: PropTypes.object.isRequired,
};

export default ThumbnailDisplay;
