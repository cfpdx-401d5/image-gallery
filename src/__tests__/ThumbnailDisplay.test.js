import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import ThumbnailDisplay from '../ThumbnailDisplay';

describe('ThumbnailDisplay component: ', function () {

  it('renders the ThumbnailDisplay component', () => {
    const rendered = shallow(<ThumbnailDisplay image={{ title: '_Title', url: '_url' }} deleteImage={() => {} } />);
    expect(toJson(rendered)).toMatchSnapshot();
  });

});
