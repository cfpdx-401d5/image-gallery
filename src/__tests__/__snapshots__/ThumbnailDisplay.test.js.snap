exports[`ThumbnailDisplay component:  renders the ThumbnailDisplay component 1`] = `
<div
  className="image-divs">
  <p>
    Title: 
    _title
     
  </p>
  <a
    href="_url">
    <img
      alt="cute bunny"
      className="thumbnail"
      src="_url" />
  </a>
  <button
    className="kill-button"
    onClick={[Function]}>
    Delete
  </button>
</div>
`;
